### ISSUE TITLE
`functional/prefer-immutable-types` Limit only to specific types

### ISSUE THREAD

**Issue by RebeccaStevens** [eslint-functional/eslint-plugin-functional#577]
## Suggestion

Use the same approach as https://github.com/typescript-eslint/typescript-eslint/pull/4436

Blocked until `@typescript-eslint/type-utils` v6 is released.

**Comment by zEh-** [comment]
Any update to this?

Would this fix `const f = (p: string) => ...` reporting `Error: Parameter should have an immutability of at least "ReadonlyShallow" (actual: "Unknown")`?

**Comment by RebeccaStevens** [comment]
I haven't started work on this yet but will soon.

This won't address that error you are getting. Could you give me the full type definition you are using.
Getting "Unknown" as the result suggest a bug due to an unhandled case is occurring.

**Comment by zEh-** [comment]
I get it in a bunch of places and am forced to disable enforcement altogether for now.
Happy to move this to a new issue. Just not sure what this relates to.

```ts
export const slugify = (text: string) => text.replace(...)
```

```ts
async function graphFetch(resource: `/${string}`) {
```

```ts
readonly onClick?: (ev: React.MouseEvent<HTMLElement>) => void
```

```ts
export const Switch = ({
  isActiveByDefault,
  onClick,
  title,
}: {
  readonly isActiveByDefault: boolean | undefined
  readonly onClick: (isActive: boolean) => void
  readonly title: string
}) => {
```

```ts
export const Tag = ({ children }: { readonly children: string }) => {
```

```ts
type Props = {
  readonly style?: CSSProperties
  readonly width: number
  readonly height: number
  readonly items: readonly Item[]
}

export const Bar = ({ style, width, height, items }: Props) => (
```

eslint config:
```json
{
  "plugins": ["prettier", "simple-import-sort", "functional"],
  "extends": [
    "next/core-web-vitals",
    "plugin:functional/lite",
    "plugin:no-template-curly-in-string-fix/recommended",
    "plugin:prettier/recommended"
  ],
  "rules": {
    "prettier/prettier": "error",
    // minor non-related stylistic custom rules...
  }
}
```
